src/webots/vrml/WbProtoManager.cpp

```diff
@@ -117,20 +119,26 @@ WbProtoModel *WbProtoManager::findModel(const QString &modelName, const QString
   assert(!parentFilePath.isEmpty());  // cannot find a model unless we know where to look
 
   QString protoDeclaration;
-  // check the cut buffer
-  if (mExternProtoCutBuffer && mExternProtoCutBuffer->name() == modelName)
-    protoDeclaration = mExternProtoCutBuffer->url();
 
-  // determine the location of the PROTO based on the EXTERNPROTO declaration in the parent file
-  if (protoDeclaration.isEmpty())
-    protoDeclaration = findExternProtoDeclarationInFile(parentFilePath, modelName);
+  // nodes imported from a supervisor should only check the IMPORTABLE list
+  if (!mImportedFromSupervisor) {
+    // check the cut buffer
+    if (protoDeclaration.isEmpty() && mExternProtoCutBuffer && mExternProtoCutBuffer->name() == modelName)
+      protoDeclaration = mExternProtoCutBuffer->url();
+
+    // determine the location of the PROTO based on the EXTERNPROTO declaration in the parent file
+    if (protoDeclaration.isEmpty())
+      protoDeclaration = findExternProtoDeclarationInFile(parentFilePath, modelName);
+  }
 
   // for IMPORTABLE proto nodes the declaration is in the EXTERNPROTO list, nodes added with add-node follow a different pipe
   if (protoDeclaration.isEmpty()) {
     foreach (const WbExternProto *proto, mExternProto) {
       if (proto->name() == modelName && (proto->isImportable() || proto->isFromRootNodeConversion()))
         protoDeclaration = proto->url();
     }
+    if (mImportedFromSupervisor && protoDeclaration.isEmpty())
+      assert(false);  // for supervisor imported nodes, there should always be a corresponding PROTO in the IMPORTABLE list
```

Comment:
It is better to encapsulate this inside the assert (to ensure it won't affect the release version):
```suggestion
    // for supervisor imported nodes, there should always be a corresponding PROTO in the IMPORTABLE list
    assert(!mImportedFromSupervisor || !protoDeclaration.isEmpty());
```